### PR TITLE
docs: revise upgrade doc for Rook v1.8

### PR DESCRIPTION
**Pull request by BlaineEXE**
Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/ceph-upgrade.md [hunk 1]
```diff
@@ -39,6 +40,16 @@ those releases.
 * [Upgrade 0.6 to 0.7](https://rook.io/docs/rook/v0.7/upgrade.html)
 * [Upgrade 0.5 to 0.6](https://rook.io/docs/rook/v0.6/upgrade.html)
 
+## Important changes in this release
+The minimum Kubernetes version has changed to v1.16. You must update to at least Kubernetes version
```
Comment (travisn) [inline Documentation/ceph-upgrade.md:44]: nit: how about we make this a bullet list?

File: Documentation/ceph-upgrade.md [hunk 2]
```diff
@@ -39,6 +40,16 @@ those releases.
...
+v1.16 before upgrading Rook from v1.7 to v1.8.
+
```
Comment (travisn) [inline Documentation/ceph-upgrade.md:46]: Even though it's also mentioned in the ceph upgrade section below, in this section how about we also mention the upgrade off Nautilus is required?

File: Documentation/ceph-upgrade.md [hunk 3]
```diff
@@ -458,6 +441,9 @@ It's best to run `ceph config-key dump` again to verify references to
 
 See for more information, see here: https://github.com/rook/rook/issues/9185
 
+### **Important consideration for CephNFS users**
+Users of CephNFS need to take additional steps to upgrade Ceph versions. Please see the
+[NFS documentation](#ceph-nfs-crd.md) for full details.
```
Comment (travisn) [inline Documentation/ceph-upgrade.md:446]:
```suggestion
[NFS documentation](ceph-nfs-crd.md) for full details.
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/ceph-upgrade.md (hunk 2 above)
Comment (BlaineEXE) [reply Documentation/ceph-upgrade.md:46]: Sounds good. 👍

File: Documentation/ceph-upgrade.md [hunk 4]
```diff
@@ -39,6 +40,16 @@ those releases.
 * [Upgrade 0.6 to 0.7](https://rook.io/docs/rook/v0.7/upgrade.html)
 * [Upgrade 0.5 to 0.6](https://rook.io/docs/rook/v0.6/upgrade.html)
 
+## Important changes in this release
+* The minimum Kubernetes version has changed to v1.16. You must update to at least Kubernetes version
+v1.16 before upgrading Rook from v1.7 to v1.8.
+* Rook v1.8 no longer supports Ceph Nautilus (14.2.x). Nautilus users must upgrade Ceph to
```
Comment (leseb) [inline Documentation/ceph-upgrade.md:46]: Do we have a doc we could point to on how to upgrade the Ceph cluster?

Comment (BlaineEXE) [reply Documentation/ceph-upgrade.md:46]: We could link to the ceph upgrade section below 👍

File: Documentation/ceph-upgrade.md [hunk 5]
```diff
@@ -39,6 +40,16 @@ those releases.
...
+  Octopus (15.2.x) or Pacific (16.2.x) before upgrading to Rook v1.8.
+* !!! TODO: Placeholder for section about migrating FlexVolume to CSI !!!
```
Comment (subhamkrai) [inline Documentation/ceph-upgrade.md:48]: I'll add doc pointing to >1.7.9 branch.

**Review by leseb — APPROVED** [review]

**Review by travisn — APPROVED** [review]
LGTM!

**Pull request merged**